docs/question/template.md

```diff
@@ -0,0 +1,242 @@
+          <markdown>
+            # Hello, world!
+          </markdown>
+        </div>
+        ```
+
+A few special behaviors have been added to enable Markdown to work better within the PrairieLearn ecosystem, as described below.
+
+### Markdown code blocks
+
+Fenced code blocks (those using triple-backticks ` ``` `) are rendered as [`<pl-code>` elements](../elements.md#pl-code-element), which will then be rendered as usual by PrairieLearn. These blocks support specifying language and highlighted lines, which are then passed to the resulting `<pl-code>` element. Consider the following Markdown:
+
+<!-- prettier-ignore -->
+````html title="question.html"
+<markdown>
+```cpp{1-2,4}
+int i = 1;
+int j = 2;
+int k = 3;
+int m = 4;
+```
+</markdown>
+````
+
+This will be rendered to the following `<pl-code>` element (which itself will eventually be rendered to standard HTML):
+
+<!-- prettier-ignore -->
+```html
+<pl-code language="cpp" highlight-lines="1-2,4">
+int i = 1;
+int j = 2;
+int k = 3;
+int m = 4;
+</pl-code>
+```
+
+??? note "Escaping literal `<markdown>` tags"
+
+    PrairieLearn defines a special escape syntax to allow a literal `<markdown>` or `</markdown>` tag in your question. If you have `<markdown#>` or `</markdown#>` in a Markdown block, they will be rendered as `<markdown>` and `</markdown>` respectively (but will not be used to find regions of text to process as Markdown).
+
+## Using LaTeX in questions (math mode)
+
+PrairieLearn supports LaTeX equations in questions. You can view a full list of [supported MathJax commands](https://docs.mathjax.org/en/latest/input/tex/macros/index.html).
+
+Inline equations can be written using `$x^2$` or `\(x^2\)`, and display equations can be written using `$$x^2$$` or `\[x^2\]`. For example:
+
+<!-- prettier-ignore -->
+```html title="question.html"
+<p>Here is some inline math: $x^2$. Here is some display math: $$x^2$$</p>
+<p>What is the total force $F$ currently acting on the particle?</p>
+
+<markdown>
+# LaTeX works in Markdown too!
+
+$$\phi = \frac{1+\sqrt{5}}{2}$$
+</markdown>
+```
+
+### Using a dollar sign ($) without triggering math mode
+
+Dollar signs by default denote either **inline** (`$ x $`) or **display mode** (`$$ x $$`) environments.
+
+To escape either math environment, consider using PrairieLearn's `<markdown>` tag and inline code syntax.
+
+<!-- prettier-ignore -->
+```html
+<markdown>
+What happens if we use a `$` to reference the spreadsheet cell location `$A$1`?
+</markdown>
+```
+
+In scenarios where it does not make sense to use the code environment, consider disabling math entirely by
+adding the `mathjax_ignore` class to an HTML element.
+
+```html
+<div class="mathjax_ignore">
+  Mary has $5 to spend. If each apple costs $2 dollars and a banana costs $1 dollar, then how many
+  pieces of fruit can Mary get?
+</div>
+
+<div>$x = 1$ and I have <span class="mathjax_ignore">$</span>5 dollars.</div>
+```
+
+## Rendering panels from `question.html`
```

Comment:
I'm wondering if this section should be moved to the top?

Reply:
I agree, and moved this to the top.